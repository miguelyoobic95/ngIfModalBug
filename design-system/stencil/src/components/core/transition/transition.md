---
name: Transition
category: Core Components
---
## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`type`|string|`fade`|the transition type: fade, bottom, scale-up, icon, image, heading|
|`heading`|string|  |The main text in a transition, only to be used for type: fade, bottom, heading|
|`subHeading`|string|   |The sub heading of a transition, only to be used for type: heading|
|`icon`|string|   |The icon of a  icon transition, only to be used for type: icon|
|`image`|string|   |The image of a image transition, only to be used for type: image|

```yoo-transition.html
    <div class="column">
        <div class="wrapper">
            <div class="spacer">
                <yoo-transition type="fade" heading="FADE" class="accent">
                    <div >
                        <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                    </div>
                </yoo-transition>
            </div>
            <div class="spacer">
                <yoo-transition type="bottom" heading="BOTTOM">
                    <div >
                        <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                    </div>
                </yoo-transition>
            </div>
            <div class="spacer">
                <yoo-transition type="scale-up">
                    <div >
                        <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                    </div>
                </yoo-transition>
            </div>
        </div>
    <div class="wrapper">
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="light">
                <div >
                    <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="image" image="http://www.intrawallpaper.com/static/images/hd-wallpaper-download-old-car.jpg">
                <div >
                    <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="heading" heading="HEADING" sub-heading="Sub Heading" class="light">
                <div >
                    <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
    </div>
    </div>
```

```yoo-transition.css hidden
    .wrapper {
        display: flex;
        flex-direction: row;
        padding: 10px;
    }
    .spacer {
        padding: 0 10px;
    }
```

## CSS
|Type|Name|Description|
|---|---|---|
|Light|`light`|transition light|
|Accent|`accent`|transition accent|
|Danger|`danger`|transition danger|
|Success|`success`|transition success|
|Info|`info` |transition info|
|Warning|`warning`|transition warning|
|dark|`dark`|transition dark|

```yoo-transition-styles.html
    <div class="column">
        <div class="wrapper">
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="light">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="accent">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="danger">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        </div>
    <div class="wrapper">
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="success">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="info">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="warning">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
    </div>
    <div class="wrapper">
        <div class="spacer">
            <yoo-transition type="icon" icon="yo-plus" class="dark">
                <div >
                    <img src="https://i.imgur.com/uomkVIL.png" alt="Image" height="200"/>
                </div>
            </yoo-transition>
        </div>
    </div>
    </div>
```

```yoo-transition-styles.css hidden
    .wrapper {
        display: flex;
        flex-direction: row;
        padding: 10px;
    }
    .spacer {
        padding: 0 10px;
    }
```