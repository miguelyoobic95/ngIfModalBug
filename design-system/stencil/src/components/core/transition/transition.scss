@mixin size-position($width, $height, $top, $left) {
    .container {
        .overlay {
            width: $width;
            height: $height;
            top: $top;
            left: $left;
        }
    }
}

:host{
    .container {
        display: inline-block;
        position: relative;
        overflow: hidden;
            .overlay {
            display: none;
            position: absolute;
            z-index: 1000;
            background-color: cssvar(dark-20);
            color: cssvar(dark);
            opacity: 0.8;
            font-size: 2rem;
            justify-content: center;
            align-items: center;
        }
        .sub-overlay {
            display: none;
            position: absolute;
            z-index: 1000;
            background-color: transparent;
            color: cssvar(dark);
            opacity: 1;
            font-size: 2rem;
            justify-content: center;
            align-items: center;            
        }
    }
}

:host(.bottom) {
    @include size-position(100%, 50%, 50%, 0%);
}

:host(.fade) {
    @include size-position(95%, 95%, 2.5%, 2.5%);
}

:host(.icon) {
    @include size-position(100%, 100%, 0%, 0%);
    .container {
        .overlay {
            font-size: 3rem;
            background-color: transparent;
            opacity: 1;
        }
    }
}

:host(.image) {
    @include size-position(100%, 100%, 0%, 0%);
    .container {
        .overlay {
            background-color: transparent;
            opacity: 1;
        }
    }
}

:host(.heading) {
    .container {
        .overlay {
            width: 100%;
            height: 50%;
            top: 0%;
            left: 0%;
            background-color: transparent;
            opacity: 1;
        }
        .sub-overlay {
            width: 100%;
            height: 50%;
            bottom: 0%;
            left: 0%;
        }
    }
}

@mixin text-color($class, $color) {
    :host(.#{$class}) {
        .container {
           .overlay {
               color: cssvar($color);
           }
           .sub-overlay {
               color: cssvar($color);
           }
        }
    }
}

@include text-color(accent, accent);
@include text-color(dark, dark);
@include text-color(danger, danger);
@include text-color(success, success);
@include text-color(info, info);
@include text-color(warning, warning);
@include text-color(light, light);