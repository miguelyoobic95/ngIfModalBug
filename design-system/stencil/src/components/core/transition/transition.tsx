import { Component, Element, Prop } from '@stencil/core';
import { setAnimation, animations } from '../../../utils/anim';

@Component({
    tag: 'yoo-transition',
    styleUrl: 'transition.scss',
    scoped: true
})
export class YooTransitionComponent {

    @Prop() type: 'fade' | 'bottom' | 'icon' | 'image' | 'heading' | 'scale-up' = 'fade';
    @Prop() heading: string;
    @Prop() icon: string;
    @Prop() image: string;
    @Prop() subHeading: string;

     @Element() host: HTMLStencilElement;

    componentWillLoad() {
        this.host.className = this.host.className + ' ' + this.type;
    }

    componentDidLoad() {
        this.host.querySelectorAll('div')[3].setAttribute('style', 'display: flex;');
    }

    onMouseEnter() {
        if (this.type === 'scale-up') {
            setAnimation(animations.transitionScale, this.host.querySelector('.slot-container'), {open: true});
        } else {
            this.animationHandler(10, true, false, 'flex');
        }
    }

    onMouseLeave() {
        if (this.type === 'scale-up') {
            setAnimation(animations.transitionScale, this.host.querySelector('.slot-container'), {open: false});
        } else {
            this.animationHandler(200, false, true, 'none');
        }
    }

    animationHandler(timeout: number, openFade: boolean, closeBottom: boolean, displayString: string) {
        let halfHeight = this.host.querySelector('.container').clientHeight / 2;
        let overlay = this.host.querySelector('.overlay');
        let subOverlay = this.host.querySelector('.sub-overlay');
        switch (this.type) {
            case 'bottom':
                setAnimation(animations.slideVertical, overlay, {up: true, distance: halfHeight, open: !closeBottom, duration: 200});
                break;
            case 'fade':
                setAnimation(animations.fade, overlay, {open: openFade, duration: 200, opacityValue: 0.7});
                break;
            case 'icon':
                setAnimation(animations.fade, overlay, {open: openFade, duration: 200, opacityValue: 1});
                break;
            case 'image':
                setAnimation(animations.fade, overlay, {open: openFade, duration: 200, opacityValue: 1});
                break;
            case 'heading':
                setAnimation(animations.slideVertical, overlay, {up: false, distance: halfHeight, open: !closeBottom, duration: 200});
                if (subOverlay) {
                    setAnimation(animations.slideVertical, subOverlay, {up: true, distance: halfHeight, open: !closeBottom, duration: 200});
                }
                break;
        }
        setTimeout(() => {
            overlay.setAttribute('style', `display: ${displayString};`);
            if (subOverlay) {subOverlay.setAttribute('style', `display: ${displayString};`); }
        }, timeout);
    }


    render(): JSX.Element {
        return (
            <div class="container" onMouseEnter={() => this.onMouseEnter()} onMouseLeave={() => this.onMouseLeave()}>
                <div class="overlay">
                    {this.image ? <img src={this.image} alt="Image"/> : (this.icon ? <div><i class={this.icon}></i></div> : this.heading)}
                </div>
                {this.subHeading ? <div class="sub-overlay">{this.subHeading}</div> : null}
                <div class="slot-container">
                    <slot/>
                </div>
            </div>
        );
    }
}
