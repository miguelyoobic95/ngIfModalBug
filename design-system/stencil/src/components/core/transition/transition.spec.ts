import { TestWindow } from '@stencil/core/testing';
import { YooTransitionComponent } from './transition';

describe('YooTransitionComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooTransitionComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it('should have a type', async () => {
            let element;
            element = await window.load ({
                components: [YooTransitionComponent],
                html: `<yoo-transition type="fade" class="accent">
                            <div >
                                <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                            </div>
                        </yoo-transition>`
            });
            expect(element).toMatchSnapshot();
        });

        it('should have a heading', async () => {
            let element;
            element = await window.load ({
                components: [YooTransitionComponent],
                html: `<yoo-transition type="fade" heading="TEXT" class="accent">
                            <div >
                                <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                            </div>
                        </yoo-transition>`
            });
            expect(element).toMatchSnapshot();
        });

        it('should have a subheading', async () => {
            let element;
            element = await window.load ({
                components: [YooTransitionComponent],
                html: `<yoo-transition type="heading" heading="TITLE" sub-heading="SUBHEADING" class="accent">
                            <div >
                                <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                            </div>
                        </yoo-transition>`
            });
            expect(element).toMatchSnapshot();
        });

        it('should have an icon', async () => {
            let element;
            element = await window.load ({
                components: [YooTransitionComponent],
                html: `<yoo-transition type="icon" class="accent">
                            <div >
                                <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                            </div>
                        </yoo-transition>`
            });
            expect(element).toMatchSnapshot();
        });

        it('should have an image', async () => {
            let element;
            element = await window.load ({
                components: [YooTransitionComponent],
                html: `<yoo-transition type="image" class="accent">
                            <div >
                                <img src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg" alt="Image" height="200"/>
                            </div>
                        </yoo-transition>`
            });
            expect(element).toMatchSnapshot();
        });
    });
});