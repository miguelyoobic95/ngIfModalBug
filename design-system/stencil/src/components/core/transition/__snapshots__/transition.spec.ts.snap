// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooTransitionComponent Snapshots should have a heading 1`] = `
<yoo-transition
  class="accent fade hydrated"
  data-yoo-transition-host=""
  heading="TEXT"
  type="fade"
>
  
  
  
  
  <div
    class="container"
    data-yoo-transition=""
  >
    <div
      class="overlay"
      data-yoo-transition=""
    >
      TEXT
    </div>
    
    <div
      class="slot-container"
      data-yoo-transition=""
      data-yoo-transition-slot=""
    >
      
      
                            
      <div
        style="display: flex;"
      >
        
                                
        <img
          alt="Image"
          height="200"
          src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg"
        />
        
                            
      </div>
      
                        
    </div>
  </div>
</yoo-transition>
`;

exports[`YooTransitionComponent Snapshots should have a subheading 1`] = `
<yoo-transition
  class="accent heading hydrated"
  data-yoo-transition-host=""
  heading="TITLE"
  sub-heading="SUBHEADING"
  type="heading"
>
  
  
  
  
  <div
    class="container"
    data-yoo-transition=""
  >
    <div
      class="overlay"
      data-yoo-transition=""
    >
      TITLE
    </div>
    <div
      class="sub-overlay"
      data-yoo-transition=""
    >
      SUBHEADING
    </div>
    <div
      class="slot-container"
      data-yoo-transition=""
      data-yoo-transition-slot=""
      style="display: flex;"
    >
      
      
                            
      <div>
        
                                
        <img
          alt="Image"
          height="200"
          src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg"
        />
        
                            
      </div>
      
                        
    </div>
  </div>
</yoo-transition>
`;

exports[`YooTransitionComponent Snapshots should have a type 1`] = `
<yoo-transition
  class="accent fade hydrated"
  data-yoo-transition-host=""
  type="fade"
>
  
  
  
  
  <div
    class="container"
    data-yoo-transition=""
  >
    <div
      class="overlay"
      data-yoo-transition=""
    >
      
    </div>
    
    <div
      class="slot-container"
      data-yoo-transition=""
      data-yoo-transition-slot=""
    >
      
      
                            
      <div
        style="display: flex;"
      >
        
                                
        <img
          alt="Image"
          height="200"
          src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg"
        />
        
                            
      </div>
      
                        
    </div>
  </div>
</yoo-transition>
`;

exports[`YooTransitionComponent Snapshots should have an icon 1`] = `
<yoo-transition
  class="accent icon hydrated"
  data-yoo-transition-host=""
  type="icon"
>
  
  
  
  
  <div
    class="container"
    data-yoo-transition=""
  >
    <div
      class="overlay"
      data-yoo-transition=""
    >
      
    </div>
    
    <div
      class="slot-container"
      data-yoo-transition=""
      data-yoo-transition-slot=""
    >
      
      
                            
      <div
        style="display: flex;"
      >
        
                                
        <img
          alt="Image"
          height="200"
          src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg"
        />
        
                            
      </div>
      
                        
    </div>
  </div>
</yoo-transition>
`;

exports[`YooTransitionComponent Snapshots should have an image 1`] = `
<yoo-transition
  class="accent image hydrated"
  data-yoo-transition-host=""
  type="image"
>
  
  
  
  
  <div
    class="container"
    data-yoo-transition=""
  >
    <div
      class="overlay"
      data-yoo-transition=""
    >
      
    </div>
    
    <div
      class="slot-container"
      data-yoo-transition=""
      data-yoo-transition-slot=""
    >
      
      
                            
      <div
        style="display: flex;"
      >
        
                                
        <img
          alt="Image"
          height="200"
          src="https://images3.alphacoders.com/823/thumb-1920-82317.jpg"
        />
        
                            
      </div>
      
                        
    </div>
  </div>
</yoo-transition>
`;
